type-definitions/immutable.js.flow

```diff
@@ -27,7 +27,28 @@
  *
  * Note that Immutable values implement the `ESIterable` interface.
  */
-type ESIterable<T> = $Iterable<T,void,void>;
+
+type IteratorResult<Yield,Return> = {
+  done: true,
+  value?: Return,
+} | {
+  done: false,
+  value: Yield,
+};
+
+
+interface _Iterator<Yield,Return,Next> {
```

Comment:
No need to redefine Iterator. There is $Iterator (https://github.com/facebook/flow/blob/master/lib/core.js#L401)